src/lib/components/globalhealthbar/GlobalHealthBar.component.js

```diff
@@ -0,0 +1,167 @@
+//@flow
+import React from 'react';
+import VegaChart from '../vegachart/VegaChart.component.js';
+import { useTheme } from 'styled-components';
+
+export const TOP = 'top';
+export const BOTTOM = 'bottom';
+type Position = typeof TOP | typeof BOTTOM;
+export type GlobalHealthProps = {
+  id: string,
+  alerts: {
+    description: string,
+    startsAt: string,
+    endsAt: string,
+    severity: string,
+  }[],
+  start: string,
+  end: string,
+  height?: number,
+  width?: number,
+  tooltipPosition?: Position,
+};
+
+function convert2TimeObject(time) {
+  // convert the date string to a date time definition object `{ date: number, month: number, year: number }`, which is required by vega-lite
+  const date = new Date(time);
+
+  return {
+    date: date.getDate(),
+    month: date.getMonth() + 1,
+    year: date.getFullYear(),
+  };
+}
+
+function GlobalHealthBar({
+  id,
+  alerts,
+  start,
+  end,
+  height = 8,
+  width = 400,
+  tooltipPosition = TOP,
+}: GlobalHealthProps) {
+  const theme = useTheme();
+
+  const startTimeObject = convert2TimeObject(start);
+  const endTimeObject = convert2TimeObject(end);
+
+  const spec = {
+    width,
+    height,
+    data: {
+      // trick: when there is no global alert, in order to display a green bar, we need to pass a non-empry array
+      values: !alerts.length ? [0] : alerts,
+    },
+    transform: [
+      {
+        calculate:
+          "datum.description !== '' ? 'View details on alert page' : ''",
+        as: 'title',
+      },
+    ],
+    layer: [
+      // Paint the entire bar with green
+      {
+        mark: { type: 'rect', cornerRadius: 15 },
+        encoding: { color: { value: theme.healthyLight } },
+      },
+      // Paint the timespan as x-axis
+      {
+        mark: { type: 'rect', cursor: 'pointer' },
+        encoding: {
+          x: {
+            field: 'startsAt',
+            type: 'temporal',
+            title: null,
+            stack: null,
+            scale: {
+              /* 
+              We have warnings from console because of this known issue.
+              https://github.com/vega/vega-lite/issues/5733
+              Note that: A date time definition object
+              https://vega.github.io/vega-lite/docs/datetime.html 
+              */
+              domain: [startTimeObject, endTimeObject],
+            },
+            axis: {
+              format: '%d %b',
+              ticks: true,
+              grid: false,
+              domainWidth: 0,
+              tickCount: 7,
+              labelExpr:
+                "[timeFormat(datum.value, '%d'), timeFormat(datum.value, '%d') == '01' ? timeFormat(datum.value, '%b') : '']",
```

Comment:
shouldn't we always display the month for the first point ? (for example a graph starting on 7 February and ending on 14th won't display any month information so I'm wondering if displaying the month on the first displayed day could help to understand that we are displaying dates here )

Reply:
Update:
![image](https://user-images.githubusercontent.com/18453133/113004318-d114cd00-9173-11eb-8362-d76a2fd99c9b.png)